libvast/test/arrow_extension_types.cpp

```diff
@@ -66,3 +70,103 @@ TEST(subnet type serde roundtrip) {
+    static_cast<const arrow::DataType&>(vast::address_extension_type())));
+  CHECK(caf::visit(
+    is_type<vast::pattern_extension_type>(),
+    static_cast<const arrow::DataType&>(vast::pattern_extension_type())));
+  CHECK(caf::visit(is_type<arrow::Int64Type, arrow::NullType>(),
+                   *arrow::int64(), *arrow::null()));
+
+  CHECK_EQUAL(caf::get_if<arrow::StringType>(&*arrow::utf8()), &*arrow::utf8());
+  CHECK(
+    caf::visit(is_type<std::shared_ptr<arrow::Int64Type>>(), arrow::int64()));
+  CHECK(caf::visit(is_type<std::shared_ptr<arrow::Int64Type>,
+                           std::shared_ptr<arrow::NullType>>(),
+                   arrow::int64(), arrow::null()));
+  auto n = arrow::null();
+  auto et = static_pointer_cast<arrow::DataType>(
+    vast::make_arrow_enum(vast::enumeration_type{{"A"}, {"B"}, {"C"}}));
+  auto pt = static_pointer_cast<arrow::DataType>(vast::make_arrow_pattern());
+  CHECK(caf::get_if<std::shared_ptr<arrow::NullType>>(&n));
+  CHECK(!caf::get_if<std::shared_ptr<arrow::Int64Type>>(&n));
+  CHECK(caf::get_if<std::shared_ptr<vast::enum_extension_type>>(&et));
+  CHECK(!caf::get_if<std::shared_ptr<vast::enum_extension_type>>(&pt));
+  CHECK(!caf::get_if<std::shared_ptr<vast::pattern_extension_type>>(&et));
+  CHECK(caf::get_if<std::shared_ptr<vast::pattern_extension_type>>(&pt));
+}
+
+template <class Builder, class T = typename Builder::value_type>
+std::shared_ptr<arrow::Array> makeArrowArray(std::vector<T> xs) {
+  Builder b{};
+  CHECK(b.AppendValues(xs).ok());
+  return b.Finish().ValueOrDie();
+}
+
+std::shared_ptr<arrow::Array> makeAddressArray() {
+  arrow::FixedSizeBinaryBuilder b{arrow::fixed_size_binary(16)};
+  return std::make_shared<vast::address_array>(vast::make_arrow_address(),
+                                               b.Finish().ValueOrDie());
+}
+
+TEST(arrow::Array sum type) {
+  auto str_arr = makeArrowArray<arrow::StringBuilder, std::string>({"a", "b"});
+  auto uint_arr = makeArrowArray<arrow::UInt64Builder>({7, 8});
+  auto int_arr = makeArrowArray<arrow::Int64Builder>({3, 2, 1});
+  auto addr_arr = makeAddressArray();
+  const auto& pattern_arr = static_cast<const arrow::Array&>(
+    vast::pattern_array(vast::make_arrow_pattern(), str_arr));
+
+  CHECK(caf::get_if<arrow::StringArray>(&*str_arr));
+  CHECK(!caf::get_if<arrow::UInt64Array>(&*str_arr));
+  CHECK(!caf::get_if<arrow::StringArray>(&*uint_arr));
+  CHECK(caf::get_if<arrow::UInt64Array>(&*uint_arr));
+  CHECK(!caf::get_if<vast::address_array>(&*uint_arr));
+  CHECK(!caf::get_if<vast::pattern_array>(&*addr_arr));
+  CHECK(caf::get_if<vast::address_array>(&*addr_arr));
+  CHECK(!caf::get_if<vast::address_array>(&pattern_arr));
+  CHECK(caf::get_if<vast::pattern_array>(&pattern_arr));
+  CHECK(caf::get_if<vast::pattern_array>(&pattern_arr));
+
+  caf::visit(is_type<arrow::StringArray>(), *str_arr);
+  caf::visit(is_type<vast::pattern_array>(), pattern_arr);
+  caf::visit(is_type<vast::pattern_array>(), *str_arr);
+
```

Comment:
Please remove whitespace within the test.

Reply:
done